Realm/ObjectServerTests/RLMPermissionsAPITests.m

```diff
@@ -804,4 +802,103 @@ - (void)testSettingUnownedRealmPermission {
     CHECK_PERMISSION_ABSENT(results, expectedPermission);
 }
 
+#pragma mark - Delete Realm upon permission denied
+
+/// A Realm which is opened improperly should report an error allowing the app to recover.
+- (void)testDeleteRealmUponPermissionDenied {
+    __block void(^workBlock)(NSError *, NSURL *url) = ^(NSError *err, __unused NSURL *url) {
+        XCTFail(@"Error handler should not be called unless explicitly expected. Error: %@", err);
+    };
+    [[RLMSyncManager sharedManager] setErrorHandler:^(NSError *error, RLMSyncSession *session) {
```

Comment:
This wrapper would make sense if it was instead something like the following:

```
__block void (^errorBlock)(NSError *, NSURL *url) = nil;
RLMSyncManager.sharedManager.errorHandler = ^(NSError *error, RLMSyncSession *session) {
    if (errorBlock) {
        errorBlock(error, session.realmURL);
        errorBlock = nil;
    }
    else {
        XCTFail(@"Error handler should not be called unless explicitly expected. Error: %@", err);
    }
};
```

This would remove the need to manually reset the block to the error case after each check.